Update links to cross-org registry sharing (#258)
hashicorp/terraform-docs-common: website/docs/cloud-docs/registry/publish-providers.mdx

### Permissions

> Private providers are only available to members of the organization where you add them. In Terraform Enterprise, they are also available to organizations that you configure to [share providers](/enterprise/admin/application/registry-sharing) with that organization.

Reviewer: ```suggestion
Users must be members of an organization to access its registry and private providers. In Terraform Enterprise, providers are also available to organizations that you configure to [share registry access](/enterprise/admin/application/registry-sharing).
```
